core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.java

```diff
@@ -400,8 +400,15 @@ public void onSwipeRight() {
     searchFiles();
     tabRecyclerView.setAdapter(tabsAdapter);
     new ItemTouchHelper(tabCallback).attachToRecyclerView(tabRecyclerView);
+
+    // Only check intent on first start of activity. Otherwise the intents will enter infinite loops
+    // when "Don't keep activities" is on.
+    if (savedInstanceState == null) {
+      handleIntentActions(getIntent());
```

Comment:
I feel like this could be a source of errors. Turn on "Don't keep activities", open a book, go to library, press device home button, use search shortcut

Reply:
(While having "Don't keep activities" active) By opening a book, going back to the library, pressing the device home button and using the search widget shortcut the app opens into the search activity with the opened book. I agree that it could be a source of bugs, but I can't find any of them yet. When `savedInstanceState != null`, the activity is guaranteed to have been re-initialized from a previously saved state. [[1](https://developer.android.com/reference/android/app/Activity#onRestoreInstanceState(android.os.Bundle))] When `savedInstaceState == null`, I can't find any guarantee for the activity to be freshly started.

Reply:
Ah well that wasn't what I was expecting, that is a good outcome. 

This is a fair sight better than it was so I'll take it